### ISSUE TITLE
Payment Processing

### ISSUE THREAD

**Issue by fmorency** [liftedinit/ghostcloud-cosmos-frontend#10]
"As a user, I need reliable and secure payment processing capabilities to facilitate smooth financial transactions within the platform."

Details TBD

**Comment by fmorency** [comment]
**Pre-Paid Gas Credits**
- Gas Credit System: Develop a system where users can buy gas credits in advance with fiat or cryptocurrency. These credits are then used to pay for transaction fees within your platform.
- Automated Gas Management: Manage gas fees on behalf of users, deducting from their pre-paid credits for each transaction.

**Comment by fmorency** [comment]
I implemented NOWPayment support in #31. It uses NOWPayment "Invoice" feature with IPN callback.

The basics work.

**Comment by fmorency** [comment]
NOWPayment is a crypto payment gateway but they also support Fiat/CC processing.

- Offer a wide range of payment methods, from global standards like Visa, Mastercard, Apple Pay, Google Pay, and Revolut Pay to regional favorites, offering flexibility and catering to your customer base.
- NOWPayment fees: https://nowpayments.io/help/about-nowpayments/about/what-are-your-fees

tl;dr: maximum 1% for crypto-to-crypto. 0.5% if no swap required.

- Credit and debit card & Apple Pay, Google Pay — 3% + 1.5 EUR
- Other bank payments — 2.5 EUR – 3.5 EUR

A user goes through KYC and pays for the purchase with fiat.

Partner receives fiat and sends crypto (crypto is sent in the currency in which the user has the first wallet in Store Outcome Wallets).

Process the crypto by crediting to Custody or to the user's wallet (to the first wallet specified in Store Outcome Wallets by default).

CC is processed by one of NOWPayment's partner
    
Minimum order limit: Minimum 6,5€ - Max 15, 000€ for both daily and monthly
